babel/number_spelling/__init__.py

```diff
@@ -0,0 +1,551 @@
+# -*- coding: utf-8 -*-
+"""Spell out numbers
+
+This could be added as a submodule to numbers, but I did not
+want to interfere with current module structure.
+
+CLDR data is not used at the moment, but it could be integrated
+later. As far as I know the data from babel/cldr/common/rbnf/
+is not used at the moment (by babel) and it would not be sufficient
+anyway (at least for Hungarian), see:
+http://cldr.unicode.org/
+
+Many structures I tried (including a singleton class inheritance
+hierarchy) but finally I decided with the solution using decorators
+because this way all relevant locale specific code displayed
+together with the logic of the single function that implements
+spelling.
+
+After the CLDR data incorporated into this module the whole
+interface could change. However to design a general
+spelling-rule-markup and implement a general speller function
+for every locale based on the markup is a serious challange.
+Any contribution is very welcome!
+
+All three context object are helper for the locale specific
+spellers. They help to write clean code for the spelling
+logic. Although they are not efficient in many ways they help
+to maintain the readability and therefore the maintainability of
+the code.
+
+:copyright: (c) 2013 by Szabolcs Blága
+:license: BSD, see LICENSE in babel for more details.
+
+The locale specific copyright messages should be placed in the locale
+specific function's docstring.
+"""
+from __future__ import division, print_function, unicode_literals
+
+import re, decimal, sys
+
+dec = decimal.Decimal
+
+
+class SpellerNotFound(Exception):
```

Comment:
I still think that exceptions should be in their own module, just like `babel.exceptions` does.